.github/workflows/build-native.yml

```diff
@@ -146,6 +146,22 @@ jobs:
             target_ext: ''
             target_upload: false
             target_container: ''
+          - os: macos-14
+            target: MACOS
+            target_bin: XCSoar
+            target_bin_path: bin
+            target_final: XCSoar
+            target_ext: ''
+            target_upload: false
+            target_container: ''
```

Comment:
Do we need to keep macos-12? i'm fine to discard it if not needed.

Reply:
I kept this at the macOS 12 GitHub actions container to still allow building for x86-64 without the need to cross-compile from an ARM architecture. I do think it's good to still be able to build for x64, given that a lot of macOS devices in use are still not ARM based yet.

Reply:
Good Point. 
Currently this one fails to compile with:
A```
pplying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/ublas_no_serialization.patch
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/no_math_throw
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/no_multiprecision
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/no_iostream
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/geometry_no_backward_compatibility
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/geometry_std_math
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/variant_no_type
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/json_no_hash
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/geometry_std_array
Now at patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/geometry_std_array
Package sdl2 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sdl2.pc'
to the PKG_CONFIG_PATH environment variable
No package 'sdl2' found
build/compile.mk:104: *** library not found: sdl2.  Stop.
gmake: 
```

Reply:
Dependencies were not getting installed correctly for OSX, fixed.